### PR TITLE
Feature/csf

### PR DESCRIPTION
**Pull request by fraincs**
Issue: 

## Summary

Storybook it moving toward CSF, and won't support anything else from Storybook 7, our tests needed to be converted.

## What I did

Converted to CSF file format (https://storybook.js.org/docs/react/api/csf) stories that were not dynamically generating stories.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -90,13 +90,15 @@
     "@sharegate/stylelint-config-recommended": "2.1.1",
     "@storybook/addon-a11y": "6.5.13-alpha.0",
     "@storybook/addon-essentials": "6.5.13-alpha.0",
+    "@storybook/addon-interactions": "^6.5.13",
```
Comment (patricklafrance) [inline package.json:93]: Is it required?

I tryed this SB feature lately and I doubt this is something we want to invest into at the moment.

File: package.json [hunk 2]
```diff
@@ -90,13 +90,15 @@
...
     "@storybook/react": "6.5.13-alpha.0 ",
+    "@storybook/testing-library": "^0.0.13",
```
Comment (patricklafrance) [inline package.json:95]: Is it required?

I tryed this SB feature lately and I doubt this is something we want to invest into at the moment.

File: packages/components/src/alert/tests/chromatic/AlertTrigger.chroma.jsx [hunk 3]
```diff
@@ -2,30 +2,28 @@ import { Alert, AlertTrigger } from "@components/alert";
...
+
+export const DefaultOpen = () => <AlertTrigger defaultOpen>
+    <Button variant="secondary">Open</Button>
+    <Alert primaryButtonLabel="Yes">
+        <Heading>Launch</Heading>
+        <Content>Are you sure you want to launch the space shuttle?</Content>
+    </Alert>
+</AlertTrigger>;
+
+Default.storyName = "default";
```
Comment (patricklafrance) [inline packages/components/src/alert/tests/chromatic/AlertTrigger.chroma.jsx:28]: I believe you should collocate the story additional parameter to it's declaration.

It would be easier for maintenance as there could be many additional parameters per story

File: packages/components/src/avatar/src/AvatarGroup.tsx [hunk 4]
```diff
@@ -58,7 +59,7 @@ function RemainingAvatars({ avatars, size, ...rest }: RemainingAvatarsProps) {
                     })}
                 </ul>
             </Tooltip>
-        </TooltipTrigger>
+        </TooltipTrigger >
```
Comment (patricklafrance) [inline packages/components/src/avatar/src/AvatarGroup.tsx:62]: Additional space which should be there before ">"

File: packages/components/src/avatar/src/AvatarGroup.tsx [hunk 5]
```diff
@@ -70,6 +71,7 @@ export function InnerAvatarGroup({
     children,
     wrap = true,
     forwardedRef,
+    remainingAvatarsProps,
```
Comment (patricklafrance) [inline packages/components/src/avatar/src/AvatarGroup.tsx:74]: Is this addition related to a specific issue?

File: packages/components/src/avatar/tests/chromatic/Avatar.chroma.jsx [hunk 6]
```diff
@@ -2,98 +2,96 @@ import { Avatar } from "@components/avatar";
...
+export const Zoom = () => <Stack>
+    <Div className="zoom-in">
+        <Avatar src={Person} name="Neil Armstrong" />
+    </Div>
+    <Div className="zoom-out">
+        <Avatar src={Person} name="Neil Armstrong" />
+    </Div>
+</Stack>;
+
+export const Styling = () => <Inline>
```
Comment (patricklafrance) [inline packages/components/src/avatar/tests/chromatic/Avatar.chroma.jsx:82]: Not a big deal. more of a personal preference but I am not a big fan of having the opening element on the same line as the dclaration as it's then not aligned with the closing element.

I would rather do

```jsx
export const Zoom = () => (
   <Stack>
       <Div className="zoom-in">
           <Avatar src={Person} name="Neil Armstrong" />
       </Div>
       <Div className="zoom-out">
           <Avatar src={Person} name="Neil Armstrong" />
       </Div>
   </Stack>
);
```

File: package.json (hunk 1 above)
Comment (fraincs) [reply package.json:93]: We had an issue with remainingAvatar that needed to be tested via interactions. On hover the remaining avatars appears and they were not styled properly.

File: packages/components/src/avatar/tests/chromatic/Avatar.chroma.jsx (hunk 6 above)
Comment (fraincs) [reply packages/components/src/avatar/tests/chromatic/Avatar.chroma.jsx:82]: Sure I feel the same, the first stories I did were like this as I copied pasted from an example ;) will fix now.

File: packages/components/src/avatar/src/AvatarGroup.tsx (hunk 5 above)
Comment (fraincs) [reply packages/components/src/avatar/src/AvatarGroup.tsx:74]: Related to the remainingAvatar issue while hovered.

File: packages/components/src/avatar/tests/chromatic/Avatar.chroma.jsx (hunk 6 above)
Comment (fraincs) [reply packages/components/src/avatar/tests/chromatic/Avatar.chroma.jsx:82]: Parenthesis are non necessary, do you prefer to have them?

File: package.json (hunk 2 above)
Comment (patricklafrance) [reply package.json:95]: Do you actually the addon-interactions for this?

My guess is that is can be done only with a play function and storybook testing library implementation.

In fact, I wonder if you really even need to use storybook testing library implementation to do it.

File: packages/components/src/avatar/src/AvatarGroup.tsx (hunk 5 above)
Comment (patricklafrance) [reply packages/components/src/avatar/src/AvatarGroup.tsx:74]: Could you elaborate please.

File: packages/components/src/avatar/tests/chromatic/Avatar.chroma.jsx (hunk 6 above)
Comment (patricklafrance) [reply packages/components/src/avatar/tests/chromatic/Avatar.chroma.jsx:82]: My personal preference is to have them but it’s not a big deal if you prefer not to

File: packages/components/src/avatar/src/AvatarGroup.tsx (hunk 5 above)
Comment (fraincs) [reply packages/components/src/avatar/src/AvatarGroup.tsx:74]: The remainingAvatarsProps has been added to support adding a `data-testid` to the remainingAvatar dom element.

File: package.json (hunk 2 above)
Comment (fraincs) [reply package.json:95]: > Do you actually the addon-interactions for this?
> 
> My guess is that is can be done only with a play function and storybook testing library implementation.
> 
> In fact, I wonder if you really even need to use storybook testing library implementation to do it.

You are right, the addon-interaction is of no use.

File: packages/components/src/avatar/src/AvatarGroup.tsx (hunk 5 above)
Comment (patricklafrance) [reply packages/components/src/avatar/src/AvatarGroup.tsx:74]: Got it! Thanks

File: packages/components/src/select/tests/jest/Select.test.tsx [hunk 7]
```diff
@@ -126,14 +126,14 @@ test("selecting an option close the menu", async () => {
         </Select>
     );
 
-    act(() => {
-        userEvent.click(getByTestId("select"));
+    await act(() => {
```
Comment (patricklafrance) [inline packages/components/src/select/tests/jest/Select.test.tsx:129]: How come you need to `await` an `act` ?

Did something change with the technology?

Comment (fraincs) [reply packages/components/src/select/tests/jest/Select.test.tsx:129]: https://github.com/testing-library/user-event/issues/1018 

https://testing-library.com/docs/user-event/convenience/#keyboard

userEvent.click now returns a promise, I have an issue where using 14.4.0 some tests are broken especially the one using userEvent.tab, adding an await fixes the issue.

File: package.json [hunk 8]
```diff
@@ -91,6 +91,7 @@
     "@storybook/addon-a11y": "6.5.13-alpha.0",
     "@storybook/addon-essentials": "6.5.13-alpha.0",
     "@storybook/react": "6.5.13-alpha.0 ",
+    "@storybook/testing-library": "^0.0.13",
```
Comment (alexasselin008) [inline package.json:94]: I tought you said you were going to remove this

File: packages/components/src/avatar/src/AvatarGroup.tsx [hunk 9]
```diff
@@ -9,6 +9,7 @@ import { Tooltip, TooltipTrigger } from "../../tooltip";
 const DefaultElement = "div";
 
 export interface InnerAvatarGroupProps extends Omit<AbstractGroupProps<typeof DefaultElement>, "gap" | "orientation"> {
+    remainingAvatarsProps?: any;
```
Comment (alexasselin008) [inline packages/components/src/avatar/src/AvatarGroup.tsx:12]: Note for myself: 
Is this prop ok? was there a comment about it earlier?

The typing is wrong here

File: packages/components/src/avatar/tests/chromatic/AvatarGroup.chroma.jsx [hunk 10]
```diff
@@ -1,205 +1,228 @@
...
+        <Avatar name="Christa McAuliffe" />
+        <Avatar name="Neil Armstrong" />
+    </AvatarGroup>
+;
+
+WithRemainingsHover.play = async () => {
+    userEvent.hover(screen.getByTestId("remaining-avatars"));
+};
+
+Default.storyName = "default";
```
Comment (alexasselin008) [inline packages/components/src/avatar/tests/chromatic/AvatarGroup.chroma.jsx:222]: why are those storyName stacked at the bottom instead of directly after the elements. All the files before this one write the storyName right after the component

File: packages/components/src/badge/tests/chromatic/BadgeCount.chroma.jsx [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+            <Text>100</Text>
+        </SquareBadge>
+        <SquareBadge style={{ border: "1px solid red" }}>
+            <Text>100</Text>
+        </SquareBadge>
+    </Inline>
+};
+
+Default.storyName = "default";
+Overlap.storyName = "overlap";
```
Comment (alexasselin008) [inline packages/components/src/badge/tests/chromatic/BadgeCount.chroma.jsx:76]: why are those storyName stacked at the bottom instead of directly after the elements. All the files before this one write the storyName right after the component

File: packages/components/src/badge/tests/chromatic/BadgeDot.chroma.jsx [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+export const Styling = {
+    render: () => <Inline>
+        <SquareBadge variant="dot" border="warning-7" />
+        <SquareBadge variant="dot" className="border-red" />
+        <SquareBadge variant="dot" style={{ border: "1px solid red" }} />
+    </Inline>
+};
+
+Default.storyName = "default";
+Overlap.storyName = "overlap";
```
Comment (alexasselin008) [inline packages/components/src/badge/tests/chromatic/BadgeDot.chroma.jsx:33]: why are those storyName stacked at the bottom instead of directly after the elements. All the files before this one write the storyName right after the component

File: packages/components/src/button/tests/chromatic/ButtonGroup.chroma.jsx [hunk 13]
```diff
@@ -3,165 +3,112 @@ import { Button, ButtonGroup, IconButton } from "@components/button";
...
             <Button variant="secondary">Cutoff</Button>
             <Button variant="secondary">Cutoff</Button>
         </ButtonGroup>
-    )
-    .add("align", () =>
+    </Stack>;
+
+Default.storyName = "default";
+
+export const ButtonGroupIconButton = () =>
```
Comment (alexasselin008) [inline packages/components/src/button/tests/chromatic/ButtonGroup.chroma.jsx:28]: why is this prefixed with ButtonGroup?

File: package.json (hunk 8 above)
Comment (fraincs) [reply package.json:94]: Oopsie, removed now.

File: packages/components/src/button/tests/chromatic/CrossButton.chroma.jsx [hunk 14]
```diff
@@ -1,27 +1,26 @@
...
+
+export const Condensed = () =>
+    <Inline alignY="end">
+        <CrossButton condensed size="2xs" aria-label="Clear" />
+        <CrossButton condensed size="xs" aria-label="Clear" />
+        <CrossButton condensed size="sm" aria-label="Clear" />
+        <CrossButton condensed aria-label="Clear" />
+    </Inline>;
+
+Default.storyName = "default";
```
Comment (alexasselin008) [inline packages/components/src/button/tests/chromatic/CrossButton.chroma.jsx:25]: why at the end here also? should be directly after the component

File: packages/components/src/avatar/tests/chromatic/AvatarGroup.chroma.jsx (hunk 10 above)
Comment (fraincs) [reply packages/components/src/avatar/tests/chromatic/AvatarGroup.chroma.jsx:222]: This is an ommission

File: packages/components/src/badge/tests/chromatic/BadgeDot.chroma.jsx (hunk 12 above)
Comment (fraincs) [reply packages/components/src/badge/tests/chromatic/BadgeDot.chroma.jsx:33]: an omission.

File: packages/components/src/date-input/tests/jest/DateInput.test.tsx [hunk 15]
```diff
@@ -328,14 +328,14 @@ describe("compact presets", () => {
             />
         );
 
-        act(() => {
-            userEvent.click(container.querySelector(":scope [aria-label=\"Date presets\"]"));
+        await act(() => {
```
Comment (alexasselin008) [inline packages/components/src/date-input/tests/jest/DateInput.test.tsx:331]: this file should be reverted

File: packages/components/src/date-input/tests/jest/DateRangeInput.test.tsx [hunk 16]
```diff
@@ -496,14 +496,14 @@ describe("compact presets", () => {
             />
         );
 
-        act(() => {
-            userEvent.click(container.querySelector(":scope [aria-label=\"Date presets\"]"));
+        await act(() => {
```
Comment (alexasselin008) [inline packages/components/src/date-input/tests/jest/DateRangeInput.test.tsx:499]: this file should be reverted

File: packages/components/src/dialog/tests/chromatic/Dialog.chroma.jsx [hunk 17]
```diff
@@ -7,420 +7,511 @@ import { Div } from "@components/html";
...
+            <Paragraph>This year, the National Science Foundation (NSF) said farewell to the iconic Arecibo Observatory in Puerto Rico after two major cable failures led to the radio telescope's collapse. The 57-year old structure was once the largest radio dish telescope in the world, and researchers have used its capabilities to make significant breakthroughs in astronomy. The Arecibo Observatory also served as the dramatic backdrop to films like "Contact" and "Goldeneye."</Paragraph>
+            <Paragraph>The facility suffered two cable failures this year, and then in early December, the suspended platform above the radio dish came crashing down.</Paragraph>
+            <Paragraph>The news about Arecibo's structural damage and subsequent decommissioning was disheartening for the local community, too. Field trips to its visitors' center are a ''rite of passage'' for Puerto Rican children.</Paragraph>
+        </Content>
+    </Dialog>
+);
+
+ImageTooBig.storyName = "image too big";
+
+export const DialogueIllustration = () => (
```
Comment (alexasselin008) [inline packages/components/src/dialog/tests/chromatic/Dialog.chroma.jsx:167]: Dialog

File: packages/components/src/button/tests/chromatic/ButtonGroup.chroma.jsx (hunk 13 above)
Comment (fraincs) [reply packages/components/src/button/tests/chromatic/ButtonGroup.chroma.jsx:28]: Because IconButton is imported, it's an Orbit component

File: packages/components/src/listbox/tests/chromatic/Listbox.chroma.jsx [hunk 18]
```diff
@@ -5,20 +5,96 @@ import { Inline, Stack } from "@components/layout";
...
-    .add("only items", () =>
-        <Listbox aria-label="Planets">
+
+export default {
+    title: "Chromatic/Listbox",
+    component: Listbox,
+    parameters: {
+        chromatic: {
+            delay: 100,
+            chromaticPauseAnimationAtEnd: true
```
Comment (alexasselin008) [inline packages/components/src/listbox/tests/chromatic/Listbox.chroma.jsx:15]: est-ce que c'est une erreur de copie pas? 

chromaticPauseAnimationAtEnd: true était pas là avant

File: packages/components/src/styling/tests/chromatic/ThemeProvider.chroma.jsx [hunk 19]
```diff
@@ -1,18 +1,19 @@
...
-            .build())
-        .build();
-}
+export default {
+    title: "Chromatic/ThemeProvider",
+    component: ThemeProvider,
+    parameters: {
+        chromatic: {
+            delay: 100,
+            chromaticPauseAnimationAtEnd: true
```
Comment (alexasselin008) [inline packages/components/src/styling/tests/chromatic/ThemeProvider.chroma.jsx:13]: hromaticPauseAnimationAtEnd: true éétiat pas la avant

**Review by alexasselin008 — APPROVED** [review]
